tests/test_utilities.py

```diff
@@ -1007,6 +1007,76 @@ def test_track_event_missing_key(self, analytics_mock):
         utils.track_event('user_id', 'event_name', 'properties')
         analytics_mock.track.assert_not_called()
 
+    @ddt.data(
+        ("2014-10-13T13:11:03Z", utils.parse_datetime("2014-10-13T13:11:03Z")),
+        (None, None)
+    )
+    @ddt.unpack
+    def test_parse_datetime_generic(self, datetime_value, expected_parsed_datetime):
+        """
+        ``parse_datetime_generic`` wraps ``parse_datetime`` such that it returns ``None`` for any bad types.
+        """
+        assert utils.parse_datetime_generic(datetime_value) == expected_parsed_datetime
+
+    @ddt.data(
+        (
+            {
+                "end": "3000-10-13T13:11:01Z",
+                "enrollment_start": "2014-10-13T13:11:03Z",
+                "enrollment_end": "2999-10-13T13:11:04Z",
+            },
+            True,
+        ),
+        (
+            {"end": None, "enrollment_start": "2014-10-13T13:11:03Z", "enrollment_end": "2999-10-13T13:11:04Z"},
+            True,
+        ),
+        (
+            {"end": "3000-10-13T13:11:01Z", "enrollment_start": None, "enrollment_end": "2999-10-13T13:11:04Z"},
+            True,
+        ),
+        (
+            {"end": "3000-10-13T13:11:01Z", "enrollment_start": "2014-10-13T13:11:03Z", "enrollment_end": None},
+            True,
+        ),
+        (
+            {
+                "end": "2014-10-13T13:11:01Z",  # end < now
+                "enrollment_start": "2014-10-13T13:11:03Z",
+                "enrollment_end": "2999-10-13T13:11:04Z",
+            },
+            False,
+        ),
+        (
+            {
+                "end": "3000-10-13T13:11:01Z",
+                "enrollment_start": "3000-10-13T13:11:03Z",  # enrollment_start > now
+                "enrollment_end": "2999-10-13T13:11:04Z",
```

Comment:
@UmanShahzad Nit: Can you swap the years for `"enrollment_start"` and `"enrollment_end"` around so that they represent a realistic scenario (where the end of the enrollment period doesn't precede the start of the enrollment period)?

Reply:
Yes, sure.